WebARonARKit/WebARonARKit/ViewController.m

```diff
@@ -876,6 +971,32 @@ - (void)userContentController:(WKUserContentController *)userContentController
             [self setShowCameraFeed:true];
         } else if ([method isEqualToString:@"hideCameraFeed"]) {
             [self setShowCameraFeed:false];
+        } else if ([method isEqualToString:@"createAnchor"]) {
+          // Construct the ARAnchor with the matrix provided from the js side.
+          NSString* jsAnchorId = params[0];
+          matrix_float4x4 modelMatrix;
+          float* pModelMatrix = (float*)(&modelMatrix);
+          for (int i = 0; i < 16; i++) {
+            pModelMatrix[i] = [params[i + 1] floatValue];
+          }
+          ARAnchor* anchor = [[ARAnchor alloc] initWithTransform:modelMatrix];
+          [self.session addAnchor:anchor];
+          // Create an entry to convert from the js id to the objective c id (and viceversa)
+          [jsAnchorIdsToObjCAnchorIds setValue:anchor.identifier.UUIDString forKey:jsAnchorId];
+          [objCAnchorIdsToJSAnchorIds setValue:jsAnchorId forKey:anchor.identifier.UUIDString];
+          // Store the anchor
+          [anchors setValue:anchor forKey:jsAnchorId];
+        } else if ([method isEqualToString:@"removeAnchor"]) {
+          // Retrive the ARAnchor from the jsAnchorId and remove it from the
+          // session. Of course, also remove all the id mapping and the anchor
+          // from the anchors container.
+          NSString* jsAnchorId = params[0];
+          ARAnchor* anchor = anchors[jsAnchorId];
+          NSString* objCAnchorId = anchor.identifier.UUIDString;
+          [jsAnchorIdsToObjCAnchorIds removeObjectForKey:jsAnchorId];
+          [objCAnchorIdsToJSAnchorIds removeObjectForKey:objCAnchorId];
+          [anchors removeObjectForKey:jsAnchorId];
+          [self.session removeAnchor:anchor];
```

Comment:
Should we dispatch an anchors removed event here? Even though someone is explicitly removing it, there might be other listeners on the VRDisplay that want to know the anchor was removed.

Reply:
The anchors created by users should only be removed explicitly by the user.

Reply:
Yes, but some other code might want to react to an anchor being removed. For example, one system manages the anchors and another visualizes them (and registers listeners on the VRDisplay).

Reply:
I leave this decision up to you. We can always add it if we have a use-case. You were going to check what ARKit does internally with add/remove and matching that makes sense.